sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Statistics.scala

```diff
@@ -54,11 +57,29 @@ case class Statistics(
 
   /** Readable string representation for the Statistics. */
   def simpleString: String = {
-    Seq(s"sizeInBytes=$sizeInBytes",
-      if (rowCount.isDefined) s"rowCount=${rowCount.get}" else "",
+    Seq(s"sizeInBytes=${format(sizeInBytes, isSize = true)}",
+      if (rowCount.isDefined) s"rowCount=${format(rowCount.get, isSize = false)}" else "",
       s"isBroadcastable=$isBroadcastable"
     ).filter(_.nonEmpty).mkString(", ")
   }
+
+  /** Show the given number in a readable format. */
+  def format(number: BigInt, isSize: Boolean): String = {
+    val decimalValue = BigDecimal(number, new MathContext(3, RoundingMode.HALF_UP))
+    if (isSize) {
+      // The largest unit in Utils.bytesToString is TB
+      val PB = 1L << 50
+      if (number < 2 * PB) {
+        // The number is not very large, so we can use Utils.bytesToString to show it.
+        Utils.bytesToString(number.toLong)
+      } else {
+        // The number is too large, show it in scientific notation.
+        decimalValue.toString() + " B"
+      }
+    } else {
+      decimalValue.toString()
```

Comment:
With or without units, the readability is the same, right? If we make them consistent, the impl of `def format(number: BigInt)` will look much cleaner.